guides/users/custom-authentication.md

```diff
@@ -0,0 +1,174 @@
+features. This model can have any name, and Solidus can integrate with your
+existing Rails application's existing `User` model.
+
+By default, Solidus provides a [`Spree::LegacyUser` model][legacy-user] that
+offers the bare minimum functionality of a user. The model is intended to be
+replaced by another `User` model, like the one provided by the
+`solidus_auth_devise` gem.
+
+The rest of this article outlines the steps required should you decide to create
+a `User` model from scratch, use an authentication solution like
+[Devise][devise], or integrate your application's existing `User` model.
+
+Note that while your user model can have whatever name you like, this article
+uses the model `MyStore::User` for its examples. 
+
+[devise]: https://github.com/plataformatec/devise 
+[legacy-user]: https://github.com/solidusio/solidus/blob/master/core/app/models/spree/legacy_user.rb
+[solidus-auth-devise]: https://github.com/solidusio/solidus_auth_devise
+
+## Set the `Spree.user_class`
+
+No matter what gem or extension you use for your store's `User` model, your
+application's `Spree.user_class` value needs to be set. By default, Solidus sets
+the `Spree.user_class` to `Spree::LegacyUser`. 
+
+You can configure a custom `Spree.user_class` in your application's
+`/config/initializers/spree.rb` file:
+
+```ruby
+# /config/initializers/spree.rb
+Spree.user_class = "MyStore::User"
+```
+
+If you use the `solidus_auth_devise` gem, your `Spree.user_class` is set to
+`Spree::User` when you run the gem's `solidus:auth:install` generator.
+
+## Prepare your user model for Solidus
+
+Once you have set the `Spree.user_class`, you can start integrating with the
+features that are associated with the `user_class`.
+
+### Custom user generator
+
+After you have created your custom `User` model and its corresponding database
+table, you can use the `spree:custom_user` generator to generate Solidus's
+required `User` table columns and some authentication helpers:
+
+```shell
+bundle exec rails generate spree:custom_user MyStore::User
+```
+
+Then, you can migrate your database to add the Solidus-specific `User` table
+columns:
+
+```shell
+bundle exec rails db:migrate
+```
+
+If you use the `spree:custom_user` generator:
+
+- The `Spree.user_class` is updated to your specified class.
+- Authentication helpers are set up for the `solidus_frontend` and
+  `solidus_backend` views and are sent to the `ApplicationController`, making it
+  available throughout your application.
+- The `spree_current_user` method is defined and is sent to the
+  `Rails::ApplicationController` and the `Spree::Api::BaseController`, making it
+  available throughout your application.
+
+### Minimum requirements
+
+Solidus requires that your `User` model's database table includes at least the
+following columns:
+
+- `spree_api_key`: A string with a user's API key. This should be limited to 48
+  characters (`:limit => 48`). 
```

Comment:
is the text between parenthesis really needed?

Reply:
:joy: Probably not. Thanks for catching that.